src/planner/macos/profiles.rs

```diff
@@ -0,0 +1,158 @@
+use std::collections::HashMap;
+
+use crate::execute_command;
+
+#[derive(thiserror::Error, Debug)]
+pub enum LoadError {
+    #[error("Profile plist parsing error: {0}")]
+    Parse(#[from] plist::Error),
+
+    #[error("Profile discovery error: {0}")]
+    ProfileListing(#[from] crate::ActionErrorKind),
+}
+
+pub async fn load() -> Result<Policies, LoadError> {
+    let buf = execute_command(
+        tokio::process::Command::new("/usr/bin/profiles")
+            .args(["show", "-output", "stdout-xml"])
+            .stdin(std::process::Stdio::null()),
+    )
+    .await?
+    .stdout;
+
+    Ok(plist::from_reader(std::io::Cursor::new(buf))?)
+}
+
+pub type Policies = HashMap<Target, Vec<Profile>>;
+
+#[derive(serde::Deserialize, serde::Serialize, Clone, Debug, Eq, PartialEq, Hash)]
+pub enum Target {
+    #[serde(rename(deserialize = "_computerlevel"))]
+    Computer,
+    #[serde(untagged)]
+    User(String),
+}
+
+#[derive(serde::Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(rename_all = "PascalCase")]
+pub struct Profile {
+    pub profile_description: Option<String>,
+    pub profile_display_name: Option<String>,
+    pub profile_identifier: Option<String>,
+    pub profile_install_date: Option<String>,
+    #[serde(rename = "ProfileUUID")]
+    pub profile_uuid: Option<String>,
+    pub profile_version: Option<usize>,
+
+    #[serde(default)]
+    pub profile_items: Vec<ProfileItem>,
+}
+
+#[derive(serde::Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(tag = "PayloadType", content = "PayloadContent")]
+pub enum ProfileItem {
+    #[serde(rename = "com.apple.systemuiserver")]
+    SystemUIServer(SystemUIServer),
+
+    #[serde(untagged)]
+    Unknown(UnknownProfileItem),
+}
+
+#[derive(serde::Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "PascalCase")]
+pub struct UnknownProfileItem {
+    payload_type: String,
+    payload_content: plist::Value,
+}
+
+impl std::cmp::Eq for UnknownProfileItem {}
+
+#[derive(serde::Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub struct SystemUIServer {
+    pub mount_controls: Option<MountControls>,
+}
+
+#[derive(serde::Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub struct MountControls {
+    #[serde(default)]
+    pub harddisk_internal: Vec<HardDiskInternalOpts>,
+}
+
+#[derive(serde::Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub enum HardDiskInternalOpts {
+    Authenticate,
+    ReadOnly,
+    Deny,
+    Eject,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn try_parse_fail() {
```

Comment:
Might be good to add a "pass" test too.

Reply:
"fail" means like it failed the test of permitting us to mount the drive, in which case our code did exactly the right thing.